trieste/models/gpflux/interface.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+import gpflux
+import gpflow
+import tensorflow as tf
+from gpflux.models import DeepGP
+
+from ...data import Dataset
+from ...types import TensorType
+from ..interfaces import ProbabilisticModel
+from ..optimizer import Optimizer, TFOptimizer
+from ..gpflow.utils import module_deepcopy
+
+
+class GPfluxPredictor(ProbabilisticModel, tf.Module, ABC):
+    """A trainable wrapper for a GPflux deep Gaussian process model. Note that Scipy optimizer
+    is not supported for deep GP models."""
+
+    def __init__(self, optimizer: TFOptimizer | None = None):
+        """
+        :param optimizer: The optimizer with which to train the model. Defaults to
+            :class:`~trieste.models.optimizer.TFOptimizer` with :class:`~tf.optimizers.Adam` with
+            batch size 100. Note: using Scipy optimizer will raise ValueError as it is not
+            supported.
+        """
+        super().__init__()
+
+        if optimizer is None:
+            optimizer = TFOptimizer(tf.optimizers.Adam(), batch_size=100)
+
+        if isinstance(optimizer.optimizer, gpflow.optimizers.Scipy):
```

Comment:
probably there are a lot of optimizers that users might try and it wouldn't work, its probably hopeless to test it this way, to me it seems better to assert we have an instance of TFOptimizer, that class should do lower level checks...

Reply:
Ok that makes sense, I'll implement that

Reply:
Done